Makefile

```diff
@@ -31,10 +31,18 @@ OS ?= linux
 ARCH ?= amd64
 BENCH_OUTPUT ?= stdout
 EXTENSIONS ?= sync,search,scrub,metrics,lint,ui,mgmt
+UI_DEPENDENCIES := search,mgmt
 comma:= ,
+space := $(null) #
 hyphen:= -
 extended-name:=
 
+define unique =
```

Comment:
it seems to me like half the logic is in the `BUILD_LABELS` assignment statement and half in this function, and the reasoning for splitting it this way is not clear. `unique` seems to do a very specific thing and cannot be called for another purpose somewhere else. Maybe a name change for it should be done as well, after the logic is refactored.